### ISSUE TITLE
COMMIT_TRANSACTION metric not reported

### ISSUE THREAD

**Issue by jkhalack** [pravega/pravega#3171]
**Problem description**
We are performing a transactional test where transactions are being created on a stream, half of them are being committed, half of them being aborted.
There are controller log entries confirming that there are transactions being committed.
However, when we examine the reported metrics, only `OPEN_TRANSACTIONS`, `CREATE_TRANSACTION`, and `ABORT_TRANSACTION` metrics are being reported (the number of aborted transactions averages at one half of created transactions, as expected).
There is no sign of `COMMIT_TRANSACTION` metric however.
We are using commit `b51c6a8` on `r0.4` branch.
(The `COMMIT_TRANSACTION` metric was present when we used commit `26fb466`)

**Problem location**
AbstractStreamMetadataStore.java ?

**Suggestions for an improvement**
`COMMIT_TRANSACTION` metric should be reported transactions being committed.

**Comment by fpj** [comment]
This metric has been added in `master`:

https://github.com/pravega/pravega/blob/master/shared/metrics/src/main/java/io/pravega/shared/MetricsNames.java#L139

It has been introduced as part of this PR: 

https://github.com/pravega/pravega/pull/3117

This is not present in `r0.4` or earlier versions.

**Comment by RaulGracia** [comment]
Looking into the history of the `MetricsNames.java` file, it seems that `COMMIT_TRANSACTION` was there for some time ago. In this sense, on September 25th it was renamed from:
`public static final String COMMIT_TRANSACTION = "controller.transactions_committed";`
to
`public static final String COMMIT_TRANSACTION = "controller.transactions.committed";`

The commit that @jkhalack points out as the last working one, dates from October 17th. Since then, the file in which this metric was being reported (`AbstractStreamMetadataStore.java`) had 2 commits (`2f44817aa7f6d45a963442a96a7f63a31cbc9a12`, `7a355f6b41083659916b866bf949c4d1ec175487`). But I don't see direct changes on the specific point where the metric is reported. 

I see two options:
1) Consider to cherry pick PR #3117 to `r0.4` in which this metric seems to be correctly reported (if I'm not wrong, @jkhalack tested that in a k8 environment).
2) If the previous option is not suitable, we can create a specific PR to address this. 

Please, let me know if any of these options work for you @fpj @jkhalack.

**Comment by jkhalack** [comment]
@fpj The COMMIT_TRANSACTION metric existed in `r0.3` and was reported for that version:
https://github.com/pravega/pravega/blob/r0.3/shared/metrics/src/main/java/io/pravega/shared/MetricsNames.java#L79

The name continues to exist in `r0.4` (as `"controller.transactions_committed"`, and I've updated the telegraf template, so it is being parsed): 
https://github.com/pravega/pravega/blob/r0.4/shared/metrics/src/main/java/io/pravega/shared/MetricsNames.java#L103 
However, it is not being reported there.

The metric is being reported for the recent commit `f89820f` on master.

**Comment by RaulGracia** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [COMMIT_TRANSACTION metric not reported](https://trello.com/c/cbcdcvNX/57-committransaction-metric-not-reported)

**Comment by RaulGracia** [comment]
I think that I have found the problem: In commit `2f44817aa7f6d45a963442a96a7f63a31cbc9a12` (PR #2971), `CommitRequestHandler.java` removed the call to `AbstractStreamMetadataStore:commitTransaction` from its logic. However, this method contained the logic for collecting metrics for committed transactions. For this reason they are not being reported since then. I'm sure that this was removed due to some reason. So, I would also like to ask @shiveshr for his input here: we have the option to fix this specifically for `r0.4` (e.g., retaking the call to `AbstractStreamMetadataStore:commitTransaction` in `CommitRequestHandler.java` that only reports the metrics) or to cherry pick PR #3117 to `r0.4`. What is your take here?

**Comment by EronWright** [comment]
The impression that @jkhalack and I have is, #3117 will cause significant breaking changes to the metrics data, and that we don't have time to absorb that change for 0.4.  If that's true then please don't backport #3117 to 0.4. Hopefully the 0.4 fix will target the `transactions_committed` metric only.

**Comment by RaulGracia** [comment]
You are right, @EronWright. That PR contains some other changes that may require to adapt things. I will provide a fix for this specific problem only.
